### PR TITLE
Dev branch

### PR DESCRIPTION
**Pull request by rcohencyberarmor**
## Overview
1. add telemetry support
2. filter sniffer from being sniffed
3. several corrections while testing it
## Additional Information

> Any additional information that may be useful for reviewers to know 
-->

### CODE REVIEW

File: pkg/config/v1/config_data_falco_mock.go [hunk 1]
```diff
@@ -51,3 +52,24 @@ func (c *ConfigDataFalcoMock) GetClusterName() string {
 
 func (c *ConfigDataFalcoMock) SetNodeName() {
 }
+
+func (c *ConfigDataFalcoMock) SetMyNamespace() {
+}
+
+func (c *ConfigDataFalcoMock) SetMyContainerName() {
```
Comment (dwertent) [inline pkg/config/v1/config_data_falco_mock.go:59]: MIssing unittest (for covarage purpose...)

File: pkg/config/v1/config_data_falco_mock.go [hunk 2]
```diff
@@ -51,3 +52,24 @@ func (c *ConfigDataFalcoMock) GetClusterName() string {
...
+
+func (c *ConfigDataFalcoMock) GetMyNamespace() string {
+	return "myNamespace"
+}
+
+func (c *ConfigDataFalcoMock) GetMyContainerName() string {
+	return "myContName"
+}
+
+func (c *ConfigDataFalcoMock) SetBackgroundContext() {
```
Comment (dwertent) [inline pkg/config/v1/config_data_falco_mock.go:70]: MIssing unittest (for covarage purpose...)

File: pkg/conthandler/container_watcher.go [hunk 3]
```diff
@@ -58,6 +59,11 @@ func (client *ContainerClientK8SAPIServer) GetApiVersion(workload any) string {
 	return w.GetApiVersion()
 }
 
+func (client *ContainerClientK8SAPIServer) GetResourceVersion(workload any) string {
```
Comment (dwertent) [inline pkg/conthandler/container_watcher.go:62]: Missing unitest

File: pkg/conthandler/container_watcher_test.go [hunk 4]
```diff
@@ -26,6 +26,10 @@ func (client *k8sFakeClient) GetApiVersion(workload any) string {
 	return "v1"
 }
 
+func (client *k8sFakeClient) GetResourceVersion(workload any) string {
```
Comment (dwertent) [inline pkg/conthandler/container_watcher_test.go:29]: MIssing unitest

File: pkg/ebpfeng/falco_sniffer_engine.go [hunk 5]
```diff
@@ -151,16 +151,17 @@ func convertStringTimeToTimeOBJ(timestamp string) (*time.Time, error) {
...
 func parseLine(line string) (*ebpfev.EventData, error) {
 	if strings.Contains(line, "drop event occured") {
-		return ebpfev.CreateKernelEvent(nil, "", "", "", "", "", "", "drop event occurred\n"), nil
+		now := time.Now()
+		return ebpfev.CreateKernelEvent(&now, "", "", "", "", "", "", "drop event occurred\n"), nil
 	}
 	lineParts := strings.Split(line, "]::[")
 	if len(lineParts) != 8 {
-		logger.L().Error("", helpers.String("we have got unknown line format, line is ", fmt.Sprintf("%s", line)))
+		logger.L().Ctx(config.GetConfigurationConfigContext().GetBackgroundContext()).Error("", helpers.String("we have got unknown line format, line is ", fmt.Sprintf("%s", line)))
```
Comment (dwertent) [inline pkg/ebpfeng/falco_sniffer_engine.go:159]: Please fix this...

**Review by dwertent — CHANGES_REQUESTED** [review]
1. I did not see any spans.
2. Please attach a screenshot of the logs printed. From what I see in the code, the logs will not look good.

**Comment by dwertent** [comment]
Also, I'm failing to understand what in this code blocked you from delivering the first version of it for testing.
Please point me toward the bugs you had and how you fixed them.
I did not detect any unitests added that indicate a bug has been fixed.

File: pkg/conthandler/container_watcher.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rcohencyberarmor) [reply pkg/conthandler/container_watcher.go:62]: we were spoke in the future of maybe get the pod from the kubelet so in this case we are going to have another implementation for the interface ContainerClient, and a different workload interface(that will not use the API server)

File: pkg/conthandler/container_main_handler.go [hunk 6]
```diff
@@ -144,10 +144,10 @@ func (ch *ContainerHandler) startRelevancyProcess(contEvent v1.ContainerEventDat
 		go ch.getSBOM(contEvent)
```
[the remark this answers is not in the document]
Comment (rcohencyberarmor) [reply pkg/conthandler/container_main_handler.go:144]: for the first image phase, this is what I would do, in the second one I'll change it to work in a queue, I can do it in the first phase as well if you think it is blocking issue

File: pkg/conthandler/container_main_handler.go [hunk 7]
```diff
@@ -127,7 +127,7 @@ func createTicker() *time.Ticker {
 func (ch *ContainerHandler) startRelevancyProcess(contEvent v1.ContainerEventData) {
```
[the remark this answers is not in the document]
Comment (rcohencyberarmor) [reply pkg/conthandler/container_main_handler.go:127]: I spoke with @matthyx  on it, I'll add it to the PR

File: pkg/config/v1/config_data.go [hunk 8]
```diff
@@ -40,8 +47,12 @@ type ConfigData struct {
 	FalcoEbpfEngineData `json:"falcoEbpfEngine"`
 	NodeData            `json:"node"`
 	ClusterName         string `json:"clusterName"`
+	AccountID           string `json:"accountID"`
 	SnifferData         `json:"sniffer"`
 	DB                  `json:"db"`
+	MyNamespace         string `json:"namespace"`
+	MyContainerName     string `json:"containerName"`
+	BackgroundContext   context.Context
```
[the remark this answers is not in the document]
Comment (rcohencyberarmor) [reply pkg/config/v1/config_data.go:55]: in general as I see it we are getting the config from several places (not just from the config map, from env vars, etc). so the config is gathering all the config from all configuration "places", therefore I added it to the config_data file. the BackgroundContext is configure for the OTEL telemetry. I can create new package for any configuration flow, your call.

**Pull request merged**